conn/conn.go

```diff
@@ -84,6 +84,18 @@ type Endpoint interface {
 	SrcIP() netip.Addr
 }
 
+// PeerAwareEndpoint is an optional Endpoint specialization for integratons that want to
+// know about the outcome of cryptorouting identification.
+//
+// If they receive a packet from a source they had not pre-identified, to learn the
+// identification WireGuard can derive from the session or handshake.
+//
+// If GetPeerEndpoint returns nil, WireGuard will be unable to respond to the peer
+// until a new endpoint is written by a later packet.
+type PeerAwareEndpoint interface {
```

Comment:
Not going to make this a struct type instead to make the type check cheaper?

The struct can embed a conn.Endpoint and then make GetPeerEndpoint be a func field.

Or did you measure it and find it not worth it?